### ISSUE TITLE
Metric compute grouping silently corrupts scores across batches

### ISSUE THREAD

**Issue by sylvesterkaczmarek** [alan-turing-institute/ice-station-zebra#537]
### Problem / Goal

Identical forecasts and observations can produce different reported scores depending on how samples are divided into batches.

Reproduced at `57227ac534d2950560ecc731a817e4d5e0241681` with Python 3.11.16, PyTorch 2.14.0 and TorchMetrics 1.9.0 (the version in `uv.lock`).

[BaseModel constructs all three MetricCollections](https://github.com/alan-turing-institute/icenet-mp/blob/57227ac534d2950560ecc731a817e4d5e0241681/icenet_mp/models/base_model.py#L159-L167) with the default `compute_groups=True`. Several custom metrics have identical state names but different update rules. If their states happen to match after the first batch, automatic grouping treats them as compatible and subsequent updates share the wrong accumulated quantity.

This affects IIEE versus signed SIE error (`sum_errors`, `sample_count`) and can also affect MAE versus RMSE (`sum_errors`, `count`). The failure is conditional on the initial states matching; this does not establish that every existing run is affected.

### Reproduction

Run from an installed checkout. No datasets, GPUs or network access are needed.

```python
import torch
from torchmetrics import MetricCollection

from icenet_mp.metrics import (
    IntegratedIceEdgeErrorPerForecastDay as IIEE,
    SeaIceExtentErrorPerForecastDay as SIE,
)


def score(separate_batches, compute_groups=True):
    metrics = MetricCollection(
        {"iiee": IIEE(), "sieerror": SIE()},
        compute_groups=compute_groups,
    )
    ice = torch.ones(1, 2, 1, 1, 1)
    water = torch.zeros_like(ice)
    if separate_batches:
        metrics.update(ice, water)
        metrics.update(water, ice)
    else:
        metrics.update(torch.cat([ice, water]), torch.cat([water, ice]))
    return {name: value.tolist() for name, value in metrics.compute().items()}


print(score(False))
print(score(True))
print(score(True, compute_groups=False))
```

Observed output:

```text
{'iiee': [625.0, 625.0], 'sieerror': [0.0, 0.0]}
{'iiee': [625.0, 625.0], 'sieerror': [625.0, 625.0]}
{'iiee': [625.0, 625.0], 'sieerror': [0.0, 0.0]}
```

Each lead contains one 25 km by 25 km cell per sample. Equal overprediction and underprediction should cancel in the signed mean SIE error. IIEE should remain 625 km^2. Splitting the same samples into two batches instead reports the unsigned disagreement as the signed error.

A second reproduction using errors of 1.0 and 0.5 reports RMSE **0.866025** instead of **0.790569** (`sqrt(0.625)`), while MAE remains 0.75.

Both examples were also exercised through the actual `BaseModel.training_step`, `validation_step` and `test_step`: six CPU regression cases fail on this commit and pass after disabling compute grouping.

### What does good look like?

- [ ] Disable automatic grouping for all three collections, or explicitly allow only metrics proven to share the same update rule. `compute_groups=False` is the smallest correction.
- [ ] Add regression tests for both collisions across training, validation and testing.
- [ ] Verify that scores match independent metric instances and remain unchanged when the same samples are repartitioned into batches.

This is a silent evaluation-correctness problem that can affect comparisons between runs using different batch sizes. These results are controlled reproductions, not measurements of affected production forecasts.

### Related context

[TorchMetrics compute-group behaviour](https://lightning.ai/docs/torchmetrics/stable/pages/overview.html#metriccollection).
